include/mbedtls/config.h

```diff
@@ -1137,7 +1137,7 @@
  * operation inside the library.
  *
  */
-//#define MBEDTLS_SSL_ASYNC_PRIVATE
+#define MBEDTLS_SSL_ASYNC_PRIVATE
```

Comment:
I was undecided. @sbutcher-arm said to include it by default because the place where it's useful is on servers which typically use a shared library build. See IOTSSL-1947

Reply:
So, we should note that we need to bump `SOVERSION` next time we release, or we can do it in this PR. It's nice to have it in default builds, but at the expense of ABI compatibility, I'm not sure yet.